engine/src/settings/settings.cpp

```diff
@@ -96,6 +103,54 @@ void Settings::merge(const Settings& settingsToMerge)
     }
 }
 
+static inline void saveSettingsAsJson(const Settings& settings, Stream& stream)
+{
+    stream.put('{');
+    bool first = true;
+    for (const auto& value : settings.getValues())
+    {
+        if (!first)
+        {
+            stream.put(',');
+        }
+        first = false;
+
+        char* pEnd;
+        strtod(value.second.c_str(), &pEnd);
+
+        // Use "" if second value is not number nor boolean
+        if (pEnd != value.second.c_str() + value.second.length() && (value.second != "true" && value.second != "false"))
+        {
+            stream.printf("\n\t\"{}\": \"{}\"", value.first.c_str(), value.second.c_str());
+        }
+        else
+        {
+            stream.printf("\n\t\"{}\": {}", value.first.c_str(), value.second.c_str());
+        }
+    }
+    stream.print("\n}\n");
```

Comment:
I think it is probably a better a idea to turn this into a `nlohmann::json`, and then serialize that value with `json.dump()`. Less code duplication.

Reply:
I avoided doing that because:
- Serializing the map into a nlohmann::json sounds like an expensive process with a lot of allocations just to then Serialize it again into a string, compared to just iterating the values
- That will add quotation marks to every value :/

Reply:
Regarding the first point, performance isn't really an issue here. We won't be saving the settings every frame thousands of times! So, IMO readability/maintainability is the priority here.

Regarding the second point, not necessarily; if you do something like `nlohmann::json json = 32`, it dumps you an integer, without the quotation marks. It will only put quotation marks if you do `json = "32"`, so it's just a matter of doing it like you're doing right now - checking if it is a number of a boolean

Reply:
But we don't have integers we just have a map of string -> string so every variable is a string

Reply:
Ah you're saying making a new json, iterating the map and adding each element individually, converting it into a double or a boolean?

Reply:
Sure, but if it is a number you can just use the return value of `strtod` no? And for the booleans, you can do `json = value == "true";`